lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -388,201 +378,529 @@ export class UserAgentApplication {
-          }
+      // Cache the state, nonce, and login request data
+      this._cacheStorage.setItem(Constants.loginRequest, window.location.href, this.storeAuthStateInCookie);
+      this._cacheStorage.setItem(Constants.loginError, "");
+      this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
+      this._cacheStorage.setItem(Constants.msalError, "");
+      this._cacheStorage.setItem(Constants.msalErrorDescription, "");
 
-          if (popUpWindow) {
-              popUpWindow.close();
-          }
-      }).catch((err) => {
-          this._logger.warning("could not resolve endpoints");
-          reject(err);
-      });
-  }
+      const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
+      this._cacheStorage.setItem(authorityKey, this.authority, this.storeAuthStateInCookie);
 
-  /**
-    * Used to redirect the browser to the STS authorization endpoint
-    * @param {string} urlNavigate - URL of the authorization endpoint
-    * @hidden
-    */
-  private promptUser(urlNavigate: string) {
-      if (urlNavigate && !Utils.isEmpty(urlNavigate)) {
-      this._logger.infoPii("Navigate to:" + urlNavigate);
-      window.location.replace(urlNavigate);
-    } else {
-      this._logger.info("Navigate url is empty");
-    }
+      // Build the URL to navigate to in the popup window
+      const urlNavigate = authenticationRequest.createNavigateUrl(scopes)  + Constants.response_mode_fragment;
+      window.renewStates.push(authenticationRequest.state);
+      window.requestType = Constants.login;
+
+      // Register callback to capture results from server
+      // TODO: Need to possible rework functionality here
+      this.registerCallback(authenticationRequest.state, scope, resolve, reject);
+
+      // Navigate url in popupWindow
+      if (popUpWindow) {
+        this._logger.infoPii("Navigated Popup window to:" + urlNavigate);
+        popUpWindow.location.href = urlNavigate;
+      }
+    }, () => {
+      // Endpoint resolution failure error
+      this._logger.info(ErrorCodes.endpointResolutionError + ":" + ErrorDescription.endpointResolutionError);
+      this._cacheStorage.setItem(Constants.msalError, ErrorCodes.endpointResolutionError);
+      this._cacheStorage.setItem(Constants.msalErrorDescription, ErrorDescription.endpointResolutionError);
+
+      // What is this? Is this the reject that is passed in?? -- REDO this in the subsequent refactor, passing reject is confusing
+      if (reject) {
+        reject(ErrorCodes.endpointResolutionError + ":" + ErrorDescription.endpointResolutionError);
+      }
+
+      // Close the popup window
+      if (popUpWindow) {
+        popUpWindow.close();
+      }
+    }).catch((err) => {
+      // All catch - when is this executed? Possibly when error is thrown, but not if previous function rejects instead of throwing
```

Comment:
why is this a warning not an error

Reply:
Logging and telemetry will need to be reviewed at a later date when major changes have been merged.